operator/pkg/resources/resources_unit_tests/ingress_test.go

```diff
@@ -0,0 +1,319 @@
+package resources_unit_tests
+
+import (
+	"context"
+	"testing"
+
+	cmmetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	vectorizedv1alpha1 "github.com/redpanda-data/redpanda-operator/operator/api/vectorized/v1alpha1"
+	"github.com/redpanda-data/redpanda-operator/operator/pkg/resources"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestRenderIngressPandaProxy(t *testing.T) {
+	tests := []struct {
```

Comment:
Just my two cents, I find that table tests become less effective the more parameters that they have. 

A golden style test here is MUCH easier to write, read, and maintain.